src/lib.rs

```diff
@@ -128,48 +134,51 @@ impl Pidlock {
             }
         }
 
-        fs::remove_file(self.path.clone()).unwrap();
+        fs::remove_file(self.path.clone()).map_err(|_err| PidlockError::IOError)?;
 
         self.state = PidlockState::Released;
         Ok(())
     }
 
     /// Gets the owner of this lockfile, returning the pid. If the lock file doesn't exist,
     /// or the specified pid is not a valid process id on the system, it clears it.
-    pub fn get_owner(&self) -> Option<u32> {
+    pub fn get_owner(&self) -> Result<Option<u32>, PidlockError> {
         let mut file = match fs::OpenOptions::new().read(true).open(self.path.clone()) {
             Ok(file) => file,
             Err(_) => {
-                return None;
+                return Ok(None);
             }
         };
 
         let mut contents = String::new();
         if file.read_to_string(&mut contents).is_err() {
             warn!(
                 "Removing corrupted/invalid pid file at {}",
-                self.path.to_str().unwrap()
+                self.path.to_str().unwrap_or("<invalid>")
```

Comment:
Does it take a string as an error message here?

Reply:
In this case, if `self.patch` can't be converted into a valid utf-8 string (whyyyyyyy), we put `"<invalid>"` there instead of the path.